app/code/Magento/Contact/Controller/Index/Post.php

```diff
@@ -33,21 +34,29 @@ class Post extends \Magento\Contact\Controller\Index
     private $mail;
 
     /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
      * @param Context $context
      * @param ConfigInterface $contactsConfig
      * @param MailInterface $mail
      * @param DataPersistorInterface $dataPersistor
+     * @param LoggerInterface $logger
      */
     public function __construct(
         Context $context,
         ConfigInterface $contactsConfig,
         MailInterface $mail,
-        DataPersistorInterface $dataPersistor
+        DataPersistorInterface $dataPersistor,
```

Comment:
@JamesonNetworks  oh sorry, my fault 
I didn't notice that this parameter has already been there before.
` DataPersistorInterface $dataPersistor`

Sorry, for misleading you.
I was wrong. No need to make it optional.